### ISSUE TITLE
OAK CD-ROM Driver Configs

### ISSUE THREAD

**Issue by empsmoke** [joncampbell123/dosbox-x#578]
Dear Jon,
Do you know how to tamper with the OAK CD-ROM drivers to get a CD-ROM drive working?
If so, I really know how to.

**Comment by joncampbell123** [comment]
If you mounted the ISO image to a CD-ROM drive AND specified which IDE port before booting DOS/Windows 95, the OAK driver should detect it as an ATAPI device.

**Comment by joncampbell123** [comment]
IMGMOUNT takes an -ide parameter that specifies which IDE port to attach to and whether IDE master or IDE slave.

Try enabling IDE primary and IDE secondary interfaces in dosbox.conf and then mount with -ide 2m to attach the ISO image to the secondary master IDE interface.

**Comment by joncampbell123** [comment]
The OAK SYS drivers probably also take some number of command line parameters in CONFIG.SYS to describe what to look for and how. Please look at those as well.

**Comment by empsmoke** [comment]
But can you integrate a configuration program, for people who don't understand or experience glitches in the emulator's built-in settings menu.

**Comment by joncampbell123** [comment]
No, I can't. When booting a guest OS DOSBox-X cannot read/write files within it.

There are probably many versions of the OAK CD-ROM drivers as well.

You may want to make an online guide instead on how to edit your CONFIG.SYS to use the drivers.

**Comment by joncampbell123** [comment]
If you like, there are free compilers for DOS (C++, Pascal, etc) you can use if you would like to write your own configuration utility for OAK CD-ROM driver installation from within the emulator. As a bonus, your utility would work in any other emulator (Bochs, VirtualBox) as well as real PC hardware.

All you have to do is support copying the OAKCDROM.SYS file into place, reading CONFIG.SYS for existing config (if any), accepting options on the command line on how to configure it (if any), and then writing CONFIG.SYS back with the updated settings.

I think it would be good programming practice for you, it isn't hard to do at all.
